spec/system/viewing_user_menu_spec.rb

```diff
@@ -5,6 +5,25 @@
 
   let(:user_menu) { PageObjects::Components::UserMenu.new }
 
+  describe "with notification limit set via plugin api" do
+    it "only displays as many notifications as the limit" do
+      sign_in(user)
+
+      visit("/latest")
+
+      3.times { Fabricate(:notification, user: user) }
+      page.execute_script <<~JS
+        require("discourse/lib/plugin-api").withPluginApi("1.22.0", (api) => {
```

Comment:
System tests are meant to simulate a user interacting with the application so I don’t think testing the client side plugin API this way is ideal. Is there a reason we didn’t write a JS acceptance test for this?

Reply:
Yes, because in JS-land it's quite tricky to test with pretenders. You have to basically stub the response to return the N records, rather than actually testing if the limit is respected. This is a way _more_ effective test than any other way to test it IMO

Reply:
In JS land, you can just assert that the request to the to server is called with the right limit parameter. That should be fairly straightforward.

Reply:
An existing example of us using system specs for testing JS APIs would be [ember_deprecation_spec.rb](https://github.com/discourse/discourse/blob/07caa5bc03eca4146b8441f3a78fa4826981d313/spec/system/ember_deprecation_spec.rb)